### PR TITLE
Add missing projects to weekly test

### PR DESCRIPTION
**Pull request by zxhdaze**
Add missing projects to weekly test, reordered some tests to match [the team repositories](https://github.com/orgs/canonical/teams/soleng-reviewers/repositories?page=1). Update all branch names to main. Please don't merge until [39](https://github.com/canonical/juju-backup-all/pull/39) and [BSENG-2322](https://warthogs.atlassian.net/browse/BSENG-2322) are completed.

[BSENG-2322]: https://warthogs.atlassian.net/browse/BSENG-2322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/weekly_tests.yaml [hunk 1]
```diff
@@ -15,6 +15,9 @@ jobs:
       fail-fast: false
       matrix:
         include:
+          - repo: bootstack-actions
+            workflow_file_name: pull-request.yaml
+            branch: main
```
Comment (dashmage) [inline .github/workflows/weekly_tests.yaml:20]: For bootstack-actions, we use `pull-request.yaml` as a reusable workflow file in other repos and is not something that needs to be run weekly.

File: .github/workflows/weekly_tests.yaml [hunk 2]
```diff
@@ -72,20 +75,32 @@ jobs:
...
-            workflow_file_name: pr.yaml
+          - repo: hardware-observer-operator
+            workflow_file_name: check.yaml
+            branch: master
+          - repo: hardware-observer-operator
+            workflow_file_name: cos_integration.yaml
+            branch: master
+          - repo: layer-beats-base 
+            workflow_file_name: ci.yaml
+            branch: master
```
Comment (dashmage) [inline .github/workflows/weekly_tests.yaml:89]: The `ci.yaml` for layer-beats-base only contains lint and unit tests. So I don't think it deserves to be run weekly.

**Review by dashmage — CHANGES_REQUESTED** [review]
We should skip adding bootstack-actions, layer-beats-base. Everything else LGTM :)

**Review by rgildein — APPROVED** [review]
LGTM, thanks

**Review by aieri — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]

**Review by dashmage — APPROVED** [review]

**Pull request merged**